packages/app/src/server/models/bookmark-folder.ts

```diff
@@ -0,0 +1,99 @@
+import { Ref, IUser } from '@growi/core';
+import { isValidObjectId } from '@growi/core/src/utils/objectid-utils';
+import ExtensibleCustomError from 'extensible-custom-error';
+import {
+  Types, Document, Model, Schema,
+} from 'mongoose';
+
+
+import loggerFactory from '../../utils/logger';
+import { getOrCreateModel } from '../util/mongoose-utils';
+
+const logger = loggerFactory('growi:models:bookmark-folder');
+
+export class InvalidParentBookmarkFolder extends ExtensibleCustomError {}
+
+
+export type IBookmarkFolderDocument = {
+  name: string
+  owner: Ref<IUser>
+  parent?: string
+}
+export interface BookmarkFolderDocument extends Document {
+  _id: Types.ObjectId
+  name: string
+  owner: Types.ObjectId
+  parent?: Types.ObjectId | null
+}
+
+export interface BookmarkFolderModel extends Model<BookmarkFolderDocument>{
+  createByParameters(params: IBookmarkFolderDocument): IBookmarkFolderDocument
+  findParentFolderByUserId(user: Types.ObjectId | string): IBookmarkFolderDocument[]
+  findChildFolderById(parentBookmarkFolder: Types.ObjectId | string): Promise<IBookmarkFolderDocument[]>
+  deleteFolderAndChildren(bookmarkFolderId: string): {deletedCount: number}
+  updateBookmarkFolder(bookmarkFolderId: string, name: string, parent: string): BookmarkFolderDocument | null
+}
+
+const bookmarkFolderSchema = new Schema<BookmarkFolderDocument, BookmarkFolderModel>({
+  name: { type: String },
+  owner: { type: Schema.Types.ObjectId, ref: 'User', index: true },
+  parent: { type: Schema.Types.ObjectId, ref: 'BookmarkFolder', required: false },
+});
+
+
+bookmarkFolderSchema.statics.createByParameters = async function(params: IBookmarkFolderDocument): Promise<BookmarkFolderDocument> {
+  const { name, owner, parent } = params;
+  let bookmarkFolder;
+
+  if (parent === null) {
+    bookmarkFolder = await this.create({ name, owner, parent:  null }) as unknown as BookmarkFolderDocument;
+  }
+  else {
+    // Check if parent folder id is valid and parent folder exists
+    const isParentFolderIdValid = isValidObjectId(parent);
+    const parentFolder = await this.findById(parent);
+
+    if (!isParentFolderIdValid || parentFolder == null) {
+      throw new InvalidParentBookmarkFolder("Parent folder id is invalid or parent folder doesn't exists");
+    }
```

Comment:
parentFolder should be nullable.
If you have any questions, please ask me.

Reply:
we can create a new folder even if the parentFolder is null. 

so when parentFolder is null, I think the error should not be thrown.

 (parentFolder != null && !isParentFolderIdValid)
i think the err should be thrown in the above case

Reply:
if you have any problems, please share.